VideoLocker/src/main/java/org/edx/mobile/discussion/DiscussionAPI.java

```diff
@@ -69,29 +70,19 @@ public void failure(RetrofitError error) {
     Context context;
 
     final DiscussionService discussionService;
+
     @Inject
-    public DiscussionAPI(Context context, Config config) {
+    public DiscussionAPI(Context context, Config config, OkHttpClient client) {
         this.context = context;
         this.config = config;
-        Gson gson = new GsonBuilder().setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES)
+        Gson gson = new GsonBuilder()
+                .setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES)
                     .setDateFormat("yyyy-MM-dd'T'HH:mm:ssZ").create();
 
-
-        OkHttpClient oauthBasedClient = new OkHttpClient();
-        File cacheDirectory = new File(context.getFilesDir(), "http-cache");
-        if (!cacheDirectory.exists()) {
-            cacheDirectory.mkdirs();
-        }
-        Cache cache = new com.squareup.okhttp.Cache(cacheDirectory, cacheSize);
-        oauthBasedClient.setCache(cache);
-      //  oauthBasedClient.interceptors().add(new GzipRequestInterceptor());
-        oauthBasedClient.interceptors().add(new OauthHeaderRequestInterceptor(context));
-        oauthBasedClient.interceptors().add(new LoggingInterceptor());
         RestAdapter restAdapter = new RestAdapter.Builder()
-                .setClient(new OkClient(oauthBasedClient))
+                .setClient(new OkClient(client))
                 .setEndpoint(config.getApiHostURL())
                 .setConverter(new GsonConverter(gson))
-          //      .setRequestInterceptor(new OfflineRequestInterceptor(context, 60))
```

Comment:
I guess we don't need offline access to discussions, but we should have it
enabled for the user API I guess (unless we store/access it in the offline
profile manually).

On Fri, Sep 25, 2015 at 8:05 PM, Brian Guertin <notifications@github.com>
wrote:

> In VideoLocker/src/main/java/org/edx/mobile/discussion/DiscussionAPI.java
> <https://github.com/edx/edx-app-android/pull/372#discussion_r40439258>:
>
> >                  .setEndpoint(config.getApiHostURL())
> >                  .setConverter(new GsonConverter(gson))
> > -          //      .setRequestInterceptor(new OfflineRequestInterceptor(context, 60))
>
> I don't know, I just deleted it because it was commented. Should we be
> using it?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-app-android/pull/372/files#r40439258>.
>